src/test/java/com/jcabi/maven/plugin/UnwovenClassesTest.java

```diff
@@ -0,0 +1,124 @@
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.maven.plugin;
+
+import java.io.File;
+import org.apache.commons.io.FileUtils;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link UnwovenClasses).
+ * @author Mihai Andronache (amihaiemil@gmail.com)
+ * @version $Id$
+ * @since 0.15
+ */
+public final class UnwovenClassesTest {
+
+    /**
+     * Classes directory.
+     */
+    private static final String CLASSES = "src/test/resources/classes";
+
+    /**
+     * Unwoven classes dir.
+     */
+    private static final String UNWOVEN = "src/test/resources/unwoven";
+
+    /**
+     * Clean also before just in case <b>AfterClass</b> fails.
+     * @throws Exception If something goes wrong
+     */
+    @BeforeClass
+    public static void cleanBefore() throws Exception {
+        deleteResourceDirs();
+    }
+
+    /**
+     * UnwovenClasses can copy compiled classes to a destination directory.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void copiesUnwovenClasses() throws Exception {
+        new UnwovenClasses(
+            new File(UNWOVEN), new File(CLASSES), "process-classes"
+        ).copy();
+        MatcherAssert.assertThat(
+            new File("src/test/resources/unwoven/MyPojo.txt").exists(),
+            Matchers.is(true)
+        );
+        MatcherAssert.assertThat(
+            new File("src/test/resources/unwoven/MySecondPojo.txt").exists(),
+            Matchers.is(true)
+        );
+    }
+
+    /**
+     * UnwovenClasses can copy test classes to a destination directory.
+     * @throws Exception If something goes wrong
+     */
+    @Test
+    public void copiesUnwovenTestClasses() throws Exception {
+        new UnwovenClasses(
+            new File(UNWOVEN), new File(CLASSES), "process-test-classes"
+        ).copy();
+        MatcherAssert.assertThat(
+            new File("src/test/resources/unwoven-test/MyPojo.txt").exists(),
+            Matchers.is(true)
+        );
+        MatcherAssert.assertThat(
+            new File(
+                "src/test/resources/unwoven-test/MySecondPojo.txt"
+            ).exists(),
+            Matchers.is(true)
+        );
+    }
+
+    /**
+     * Clean resources after tests run.
+     * @throws Exception If something goes wrong
+     */
+    @AfterClass
+    public static void cleanAfter() throws Exception {
+        deleteResourceDirs();
+    }
+
+    /**
+     * Delete test resource directories (for cleanup).
+     * @throws Exception If something goes wrong
+     */
+    public static void deleteResourceDirs() throws Exception {
```

Comment:
@amihaiemil should be `private`. Normally Qulice would fail on any `public static` method, but in this project we use old version of it